secret.go

```diff
@@ -66,12 +123,87 @@ func getGitSecret(userVal string) string {
 	}
 	return gitPat
 }
-
-func createTextSecret(secretName string, identifier string, secretValue string) HarnessSecret {
+func isFileTypeSecret(secType string) bool {
+	if secType == "gcp" {
```

Comment:
Use the pre-declared enum value from constants.go instead of hard-coding here 
`GCP string = "GCP"`